### PR TITLE
Implement ThemeProvider

### PR DESCRIPTION
**Pull request by artem-kurchenko**
- Changed ThemeService to a more React-generic solution
- Fixed warning about addListener

### CODE REVIEW

**Review by dxArtemiusz — APPROVED** [review]
Ok

File: react/src/contexts/theme.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                    let startPosition = themeMarkerPosition + themeMarker.length,
+                        endPosition = href.indexOf(".css"),
+                        fileNamePart = href.substring(startPosition, endPosition);
+                    styleSheet.disabled = fileNamePart !== theme;
+                }
+            }
+        }
+        window.localStorage[storageKey] = theme;
+        currentTheme('material.' + theme);
+        refreshTheme();
```
Comment (LexDX) [inline react/src/contexts/theme.js:40]: I do not see the Swatch change. As this example is based on our Project Template, I think this is the must have feature.

**Review by 16adianay — APPROVED** [review]

**Review by LexDX — CHANGES_REQUESTED** [review]

File: react/src/contexts/theme.js (hunk 1 above)
Comment (artem-kurchenko) [reply react/src/contexts/theme.js:40]: It's a different task. Here I'm correcting React-related logic. See the original description

Comment (LexDX) [reply react/src/contexts/theme.js:40]: Ok. But we need to create such an example (i.e. improve the current one) in any way as this should cover a lot of common tasks.

Comment (artem-kurchenko) [reply react/src/contexts/theme.js:40]: yep thanks, it's in plans

**Pull request merged**